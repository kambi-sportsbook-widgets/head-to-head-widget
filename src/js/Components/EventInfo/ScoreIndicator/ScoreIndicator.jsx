import React from 'react';
import PropTypes from 'prop-types';
import styles from './ScoreIndicator.scss';

const ScoreIndicator = ({ score }) => {
   return (
      <div className={styles.container}>
         <div className={'KambiWidget-box--neutral ' + styles.scoreBox}>
            { score }
         </div>
      </div>)
};

ScoreIndicator.propTypes = {
   score: PropTypes.number.isRequired,
};

export default ScoreIndicator;
