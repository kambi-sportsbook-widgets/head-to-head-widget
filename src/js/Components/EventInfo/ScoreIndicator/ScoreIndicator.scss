:local(.container) {
   display:flex;
   flex-direction: row;
   flex-wrap: nowrap;
   align-items: center;
   flex-basis: 48px;
}

:local(.scoreBox) {
   background-color: currentColor;
   border-radius: 2px;
   text-align: center;
   width: 100%;
   height: 40px;
   min-height: 40px;
   margin: 0 2px;
}

:local(.scoreBoxText) {
   font-size: 24px;
   font-weight: bold;
   line-height: 40px;
}
