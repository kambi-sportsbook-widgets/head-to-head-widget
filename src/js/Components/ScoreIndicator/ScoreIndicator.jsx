import React from 'react';
import styles from './ScoreIndicator.scss';

const ScoreIndicator = ({ name, score, reverse }) => {

   const getContainerClass = () => {
      let name = styles.container;
      if (reverse) {
         name = styles.containerReversed;
      }
      return name;
   };

   return (
      <div className={getContainerClass()}>
         <div className={styles.name}>{ name }</div>
         <div className={styles.scoreBox}>
            <span className={styles.scoreValue}>{ score }</span>
         </div>
      </div>)
};

ScoreIndicator.propTypes = {
   score: React.PropTypes.number.isRequired,
   name: React.PropTypes.string.isRequired,
   reverse: React.PropTypes.Boolean
};

export default ScoreIndicator;
