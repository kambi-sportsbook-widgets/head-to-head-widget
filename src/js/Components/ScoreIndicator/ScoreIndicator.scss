@import '../../shared';


:local(.container) {
   display:flex;
   flex-direction: row;
   flex-wrap: nowrap;
}

:local(.containerReversed) {
   display:flex;
   flex-direction: row-reverse;
   flex-wrap: nowrap;
}

:local(.name) {
   @include smoothing();
   line-height: 40px;
   font-size: 16px;
}

:local(.scoreBox) {
   border-radius: 2px;
   text-align: center;
   background-color: #333333;
   width: 38px;
   height: 40px;
}

:local(.scoreValue) {
   @include smoothing();
   font-size: 24px;
   font-weight: bold;
   line-height: 40px;
   color: #FFFFFF;
}
