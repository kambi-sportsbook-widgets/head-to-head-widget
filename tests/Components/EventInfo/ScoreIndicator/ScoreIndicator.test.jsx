/* eslint-env jest */
import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import ScoreIndicator from '../../../../src/js/Components/EventInfo/ScoreIndicator/ScoreIndicator';

let renderer;

describe('ScoreIndicator.render()', () => {

   beforeEach(() => {
      renderer = new ReactShallowRenderer();
   });

   it('renders correctly with score', () => {
      expect(renderer.render(
         <ScoreIndicator score={100} />
      )).toMatchSnapshot();
   });

});
