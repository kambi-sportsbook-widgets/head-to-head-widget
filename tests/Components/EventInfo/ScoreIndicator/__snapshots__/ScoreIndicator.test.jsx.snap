// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreIndicator.render() renders correctly with score 1`] = `
<div
  className="container"
>
  <div
    className="KambiWidget-neutralPrimaryColor scoreBox"
  >
    <span
      className="KambiWidget-neutralSecondaryColor scoreBoxText"
    >
      100
    </span>
  </div>
</div>
`;
